### PR TITLE
Fixes #14, Adding more options to tools/pinyin

### LINKED ISSUE #14: Adding more options to tools/pinyin
- Ego version (or commit ref): `v0.40.0`
- Go version: `go1.14.3`
- Gcc version: `gcc version 9.3.0`
- Operating system and bit: `5.4.0-33-generic x86_64`

## Description

I need a special format of the `tools/pinyin` output to be:

```
$ pinyin 中国话
ZhongGuoHua
```

Justification: this is mainly to rename Chinese-named files into their pinyin counterparts. Thus

- the space is unwelcome, and because of this
- need capital each letter to easily tell them apart

I.e., two more switches will be needed/added, for space and capitalization. This would be the 99.9% use case *for me* to use this tool. 

I was planning to fork and add the feature on my own, but think I'd better ask first and see if such added output format would be welcome upstream -- I can do all the modification and send in PR myself, if idea accepted.

### PR DESCRIPTION
**Pull request by suntong**
**Please provide Issues links to:**

- Issues: #14 

**Provide test code:**

```sh
$ pinyin 中国话
zhōng guó huà 

$ pinyin -s zhao 中国话
zhong guo hua 

$ pinyin -d '' 中国话
zhōngguóhuà

$ pinyin -d '' -c 中国话
ZhōngGuóHuà

$ pinyin -s zhao -d '' -c 中国话
ZhongGuoHua

$ PINYIN_S=zhao PINYIN_D='_' pinyin 中国话
zhong_guo_hua_

$ PINYIN_S=zhao PINYIN_D='_' pinyin -c 中国话
Zhong_Guo_Hua_

 $ pinyin -h
Usage of pinyin:
  -c	Capitalize. 启用首字符大写模式
  -d string
    	Delimiter. 间隔符号 (default " ")
  -e	Heteronym. 启用多音字模式
  -help
    	Help. 显示帮助
  -p	Phrase. 启用短语
  -s string
    	Style. 指定拼音风格。可选值：zhao, zh4ao, zha4o, zhao4, zh, z, ao, 4ao, a4o, ao4 (default "zh4ao")
  -v	Verbose. 启用详细输出模式

$ pinyin -help

Usage:
 pinyin [flags..] HANS [HANS ...]

Flags:

  -s	Style. 指定拼音风格。可选值：zhao, zh4ao, zha4o, zhao4, zh, z, ao, 4ao, a4o, ao4 (PINYIN_S)
  -d	Delimiter. 间隔符号 (PINYIN_D)
  -c	Capitalize. 启用首字符大写模式 (PINYIN_C)
  -e	Heteronym. 启用多音字模式 (PINYIN_E)
  -p	Phrase. 启用短语 (PINYIN_P)
  -v	Verbose. 启用详细输出模式 (PINYIN_V)
  -help	Help. 显示帮助 (PINYIN_HELP)

Details:

  -c	Capitalize. 启用首字符大写模式
  -d string
    	Delimiter. 间隔符号 (default " ")
  -e	Heteronym. 启用多音字模式
  -help
    	Help. 显示帮助
  -p	Phrase. 启用短语
  -s string
    	Style. 指定拼音风格。可选值：zhao, zh4ao, zha4o, zhao4, zh, z, ao, 4ao, a4o, ao4 (default "zh4ao")
  -v	Verbose. 启用详细输出模式

-s 或 -d 也可以由 PINYIN_S 或 PINYIN_D 环境变量指定

```
    
## Description

Adding more options to tools/pinyin

### CODE REVIEW

File: tools/pinyin/config.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	if _, exists = os.LookupEnv("PINYIN_V"); Opts.Verbose || exists {
+		Opts.Verbose = true
+	}
+	if _, exists = os.LookupEnv("PINYIN_HELP"); Opts.Help || exists {
+		Opts.Help = true
+	}
+
+}
+
+const USAGE_SUMMARY = "  -s\tStyle. 指定拼音风格。可选值：zhao, zh4ao, zha4o, zhao4, zh, z, ao, 4ao, a4o, ao4 (PINYIN_S)\n  -d\tDelimiter. 间隔符号 (PINYIN_D)\n  -c\tCapitalize. 启用首字符大写模式 (PINYIN_C)\n  -e\tHeteronym. 启用多音字模式 (PINYIN_E)\n  -p\tPhrase. 启用短语 (PINYIN_P)\n  -v\tVerbose. 启用详细输出模式 (PINYIN_V)\n  -help\tHelp. 显示帮助 (PINYIN_HELP)\n\nDetails:\n\n"
```
Comment (vcaesar) [inline tools/pinyin/config.go:84]: Please use go style name.

**Comment by suntong** [comment]
> Please use go style name

Hmm... I don't want to guess what you mean (as I am thinking that I am already using go style names) -- could you make it clear exactly what need to be changed please?

**Comment by suntong** [comment]
Did you mean environment variables like `PINYIN_S` 或 `PINYIN_D`, etc?
I think Shell variables are better to follow Shell variables style names, especially for constants like these.

**Comment by vcaesar** [comment]
`const USAGE_SUMMARY` is not hump named.

**Comment by suntong** [comment]
Like the fix in 66d80c1? Is everything (else) OK now?

**Comment by suntong** [comment]
Or should I use `usageSummary` instead?

**Review by vcaesar — APPROVED** [review]

**Comment by suntong** [comment]
Thanks. 

Hmm... would you mind if I PR in more _"extra"_ files?

As we can see from
https://github.com/go-ego/gpy/blob/22b7fd4e33d1071a60162b98763d8181a3dc47fb/tools/pinyin/config.go#L2

The whole `config.go` is actually automatically generated, and   
there are two more code auto generation support files that I didn't push (yet), which you can see from here --
https://github.com/suntong/gpy/commit/b36aaa3a18aa684b5439472f948ed7dbd452e1e2

Would you want to include them here as well, or you'd rather to keep the repo clean than complete?

**Comment by vcaesar** [comment]
Thx. But I don’t understand the generation tool now, so there is no need to submit.

**Comment by suntong** [comment]
ok. 

the generation tool is documented at the top 3 lines of `pinyin_cli.yaml` file. I.e. --

See https://github.com/go-easygen/wireframe/wiki/Command-line-flag-handling-code-auto-generation#opensesame
for details steps of how to use it to create the cli program.

**Comment by suntong** [comment]
Hi @vcaesar, 

Would you do me a favor and add a "hacktoberfest" topic to the `gpy` and `gse` repository please? Like:

image

When I started sending PR to `gpy`, my goal was not for participating in the hacktoberfest, because I'm really not sure if I can make four acceptable PRs or not, to qualify for 

https://hacktoberfest.digitalocean.com/#rules

> To earn your Hacktoberfest tee or tree reward, you must register and make four valid pull requests (PRs) between October 1-31 (in any time zone). PRs can be made to participating public repos on GitHub, those that have the Hacktoberfest topic. If a maintainer reports your pull request as spam or behavior not in line with the project’s code of conduct, you will be ineligible to participate. This year, the first 70,000 participants who successfully complete the challenge will be eligible to receive a prize.

Now that I've made _over_ four accepted PRs, (5 in total, not counting the two turned-downed ones), I'm thinking to try my luck if I can make into that first 70,000 participants who successfully complete the challenge. 

This is the first year I know of Hacktoberfest, so it'd be a kind of honor for me if I can make it. 

Please help, thanks!

**Comment by suntong** [comment]
OK, I sense that you're reluctant to do that, participating in Hacktoberfest. I understand and respect your decision. The reason I suggested so is that it requires minimum effort from you. If you don't want to do that, that's fine with me, but could I ask that you tag each of my PR with "hacktoberfest" tag please? Like this:

image

It requires you to do a little bit more @vcaesar, but it means the recognition to the efforts that I put in, and it means a lot to me.

Thanks

**Comment by vcaesar** [comment]
I don’t know much about the event, and I add it. The reason why I don’t participate in speeches and events, is to keep a low profile for the time being, but even this has attracted a lot of abuse and slander.

**Comment by suntong** [comment]
Oh, thanks a lot for your help @vcaesar, yeah, totally understand, I get them as well, even for my own repo that nobody cares.

Hmm... I've made a mistake -- the tag for repo is "hacktoberfest", but the [tag for PR is "hacktoberfest-accepted" instead](https://hacktoberfest.digitalocean.com/details#rules). I only found it out after I investigated why my PRs are still not qualified, as this is the first year I participate Hacktoberfest. 

So I have to trouble you to tag them again, with "**hacktoberfest-accepted**". This time I verified with another PR before asking you:

image

So I'm sure this would work. 

For your convenient, here are all PRs I've made so far

- https://github.com/go-ego/gpy/pull/13
- https://github.com/go-ego/gse/pull/71
- https://github.com/go-ego/gpy/pull/18
- https://github.com/go-ego/gpy/pull/19
- https://github.com/go-ego/gse/pull/77
- https://github.com/go-ego/gpy/pull/21
- https://github.com/go-ego/gpy/pull/22


Thanks again for your help @vcaesar

**Comment by suntong** [comment]
Bump. 

Please help @vcaesar

**Comment by suntong** [comment]
Thanks a lot bro!

Could you add the following two as well pls?

- https://github.com/go-ego/gse/pull/71
- https://github.com/go-ego/gse/pull/77

thx!

**Comment by vcaesar** [comment]
OK, but adding and deleting a useless label is a hassle.

**Comment by suntong** [comment]
Oh, understand now. Sure. leave it then... 

Ops, you've already add it. Sorry to bother you. Thanks a lot for your help!

**Pull request merged**